src/Subscription.ts

```diff
@@ -66,11 +67,25 @@ export class Subscription implements ISubscription {
       return;
     }
 
-    this.closed = true;
-
-    const { _unsubscribe, _subscriptions } = (<any> this);
+    let { _parent, _parents, _unsubscribe, _subscriptions } = (<any> this);
 
-    (<any> this)._subscriptions = null;
+    this.closed = true;
+    this._parent = null;
+    this._parents = null;
+    // null out _subscriptions first so any child subscriptions that attempt
+    // to remove themselves from this subscription will noop
+    this._subscriptions = null;
+
+    let index = -1, len = _parents && _parents.length || 0;
+
+    // if this._parent is null, then so is this._parents, and we
+    // don't have to remove ourselves from any parent subscriptions.
+    while (_parent) {
+      _parent.remove(this);
+      // if this._parents is null or index >= len,
+      // then _parent is set to false, and the loop exits
+      _parent = ++index < len && _parents[index];
```

Comment:
I think this is harder to grok than neccesary--shouldn't need a comment that describes what it does.